tools/devservices_healthcheck.py

```diff
@@ -1,47 +1,183 @@
 from __future__ import annotations
 
+import argparse
 import os
 import subprocess
 import time
+from collections.abc import Sequence
+from typing import Callable
 
 
-def run_cmd(
-    args: list[str],
-    *,
-    retries: int = 3,
-    timeout: int = 5,
-) -> None:
-    for retry in range(1, retries + 1):
-        returncode = subprocess.call(args)
+class HealthcheckError(Exception):
+    pass
+
+
+class HealthCheck:
+    id: str
+    container_name: str
+    check_by_default: bool
+    check: Callable[[], None]
+    deps: list[str]
+    retries: int
+    timeout_secs: int
+
+    def __init__(
+        self, id, container_name, check_by_default, check=None, deps=None, retries=3, timeout_secs=5
+    ):
+        self.id = id
+        self.container_name = container_name
+        self.check_by_default = check_by_default
+        self.check = check
+        self.deps = deps or []
+        self.retries = retries
+        self.timeout_secs = timeout_secs
+
+    def check_container(self):
+        response = subprocess.run(
+            ("docker", "container", "inspect", "-f", "'{{.State.Status}}'", self.container_name),
+            capture_output=True,
+            text=True,
+        )
+        if response.stdout.strip() != "'running'":
+            raise HealthcheckError(f"Container '{self.container_name}' is not running.")
+
+
+def check_kafka():
+    # sentry_zookeeper:2181 doesn't work in CI, but 127.0.0.1 doesn't work locally
+    zookeeper_origin = "sentry_zookeeper"
+    if os.getenv("CI") == "true":
+        zookeeper_origin = "127.0.0.1"
```

Comment:
hmmm this makes it slightly harder to fix.  we should be trying to make CI work like local and threading this through now is going to be tricky

Reply:
I do plan to make CI work the same as local, however that is not a trivial thing.

This change means:
1. The tool can be used both locally and on CI
2. I can use the healthchecks in pytest fixtures

I'm happy to change this so it is CI only like before if you prefer.

Reply:
up to you -- either way!